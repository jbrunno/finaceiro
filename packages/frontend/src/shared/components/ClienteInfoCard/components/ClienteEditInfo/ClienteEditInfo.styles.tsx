import { css, styled } from '@mui/material/styles';
import {
  ButtonLoading,
  CardContent,
  DialogActions,
  RadioGroup,
  TextField,
} from '@frontend/components';

export const StyledForm = styled('form')(
  ({ theme }) => css`
    display: grid;
    row-gap: ${theme.fn.spacing(25)};
    width: 100%;
    min-width: ${theme.fn.size(300)};
  `,
);

export const DialogActionsStyled = styled(DialogActions)`
  padding: ${({ theme }) => theme.fn.spacing(16, 40, 48, 40)};
  width: 100%;
`;

export const RadioGroupStyled = styled(RadioGroup)(
  ({ theme }) => css`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    gap: ${theme.fn.spacing(8)};
    margin-bottom: ${theme.fn.spacing(12)};
  `,
);
export const CardHeader = styled('div')`
  padding: ${({ theme }) => theme.fn.spacing(24, 0, 32, 0)};
  text-align: center;
`;

export const CardContentStyled = styled(CardContent)`
  padding: ${({ theme }) => theme.fn.spacing(0, 40, 0, 40)};
`;

export const ButtonLoadingStyled = styled(ButtonLoading)`
  width: 100%;
`;

export const TextFieldStyled = styled(TextField)`
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`;
