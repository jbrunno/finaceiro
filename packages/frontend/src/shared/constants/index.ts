export const DEFAULT_SNACKBAR_TIMER = 4000;

export enum ClienteSexoEnum {
  MASCULINO = 'MASCULINO',
  FEMININO = 'FEMININO',
}

export const CLIENTE_SEXO = {
  [ClienteSexoEnum.MASCULINO]: 'Masculino',
  [ClienteSexoEnum.FEMININO]: 'Feminino',
};

export const OptionsSexo = Object.keys(CLIENTE_SEXO).map((key) => ({
  label: CLIENTE_SEXO[key as keyof typeof ClienteSexoEnum],
  value: key,
}));

export enum ContatoClassificacaoEnum {
  INDEFINIDO = 'INDEFINIDO',
  RUIM = 'RUIM',
  BOM = 'BOM',
  EXCELENTE = 'EXCELENTE',
}

export const CONTATO_CLASSIFICACAO = {
  [ContatoClassificacaoEnum.INDEFINIDO]: 'Indefinido',
  [ContatoClassificacaoEnum.RUIM]: 'Ruim',
  [ContatoClassificacaoEnum.BOM]: 'Bom',
  [ContatoClassificacaoEnum.EXCELENTE]: 'Excelente',
};

export const CONTATO_CLASSIFICACAO_STARS = [
  ContatoClassificacaoEnum.INDEFINIDO,
  ContatoClassificacaoEnum.RUIM,
  ContatoClassificacaoEnum.BOM,
  ContatoClassificacaoEnum.EXCELENTE,
];

export enum ContatoMarcadorEnum {
  PESSOAL = 'PESSOAL',
  TRABALHO = 'TRABALHO',
  AVOS = 'AVOS',
  FILHO = 'FILHO',
  MAE = 'MAE',
  PAI = 'PAI',
  NETO = 'NETO',
  SOBRINHO = 'SOBRINHO',
  TIO = 'TIO',
}

export const CONTATO_MARCADOR = {
  [ContatoMarcadorEnum.PESSOAL]: 'Pessoal',
  [ContatoMarcadorEnum.TRABALHO]: 'Trabalho',
  [ContatoMarcadorEnum.AVOS]: 'Avô / Avó',
  [ContatoMarcadorEnum.FILHO]: 'Filho(a)',
  [ContatoMarcadorEnum.MAE]: 'Mãe',
  [ContatoMarcadorEnum.PAI]: 'Pai',
  [ContatoMarcadorEnum.NETO]: 'Neto(a)',
  [ContatoMarcadorEnum.SOBRINHO]: 'Sobrinho(a)',
  [ContatoMarcadorEnum.TIO]: 'Tio(a)',
};

export enum ClienteTipoEnum {
  FISICA = 'FISICA',
  JURIDICA = 'JURIDICA',
  ESTRANGEIRA = 'ESTRANGEIRA',
  INDEFINIDO = 'INDEFINIDO',
}

export enum ERRORS_API {
  ERROR_401 = 401,
  ERROR_403 = 403,
  ERROR_406 = 'Esse dado já está cadastrado.',
}
