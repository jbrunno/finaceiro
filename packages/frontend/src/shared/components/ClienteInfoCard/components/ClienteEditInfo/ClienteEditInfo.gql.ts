import { ClienteSexoEnum, ClienteTipoEnum } from '@/shared/constants';
import { gql } from '@apollo/client';

export const CLIENTE_EDIT_MUTATION = gql`
  mutation Mutation($input: ClientePatchInput!) {
    clientePatch(input: $input) {
      cliente {
        cbo {
          id
          codigo
          ocupacao
        }
        tipo
        falecido
        negativado
        nomeSocial
      }
    }
  }
`;

export type ClienteEditInfoData = {
  cliente: Cliente;
};

export type ClienteEditInfoInput = {
  id: string;
  cboId?: string;
  clientMutationId?: string;
  falecido?: boolean;
  nomeSocial?: string;
  nome?: string;
  dataNascimento?: string;
  documento?: string;
  tipoCliente?: string;
};

export type Cliente = {
  id: string;
  nome: string;
  documento: string;
  nomeSocial: string;
  dataNascimento: string;
  sexo: ClienteSexoEnum;
  cbo: Cbo | null;
  falecido: boolean;
  tipo: ClienteTipoEnum;
};

export type Cbo = {
  id: string;
  codigo: string;
  ocupacao: string;
};
