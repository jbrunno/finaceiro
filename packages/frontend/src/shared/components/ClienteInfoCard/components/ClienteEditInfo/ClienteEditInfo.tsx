import {
  Box,
  Card,
  DatePicker,
  FormControlLabel,
  MenuItem,
  Radio,
  RenderComponents,
  TextField,
  Typography,
} from '@frontend/components';
import { OptionsSexo } from '@/shared/constants';
import { AutocompleteOcupacao } from '@/shared/components/AutocompleteOcupacao/AutocompleteOcupacao';
import { DateOnly } from '@frontend/utils';
import { ClienteTipoEnum } from '@frontend/types';
import {
  ButtonLoadingStyled,
  CardContentStyled,
  CardHeader,
  DialogActionsStyled,
  RadioGroupStyled,
  StyledForm,
  TextFieldStyled,
} from './ClienteEditInfo.styles';
import {
  ClienteEditFormFields,
  useClienteEditInfo,
} from './hooks/useClienteEditInfo';
import { Cliente } from './ClienteEditInfo.gql';

export type ClienteEditInfoProps = {
  cliente?: Cliente | null;
  onSuccess: () => void;
  screen: string;
};

export function ClienteEditInfo({
  onSuccess,
  screen,
  cliente,
}: ClienteEditInfoProps) {
  const { formik, maxLength, typeInput, handleChangeDocumento } =
    useClienteEditInfo({
      onSuccess,
      cliente,
    });

  return (
    <Card>
      <CardHeader>
        <Typography variant="h3">Informações básicas</Typography>
      </CardHeader>
      <CardContentStyled>
        <StyledForm onSubmit={formik.handleSubmit}>
          <RenderComponents screen={screen}>
            <TextField
              id="nomeRegistro"
              label="Nome de Registro"
              variant="outlined"
              size="small"
              value={formik.values.nome}
              onChange={formik.handleChange(ClienteEditFormFields.nome)}
              onBlur={formik.handleBlur(ClienteEditFormFields.nome)}
              error={!!(formik.touched.nome && formik.errors.nome)}
              w-perm="CLIENTE_INFO_EDIT_NOME_REGISTRO"
            />
            <TextField
              id="nomeSocial"
              label="Nome social"
              variant="outlined"
              size="small"
              value={formik.values.nomeSocial}
              onChange={formik.handleChange(ClienteEditFormFields.nomeSocial)}
              onBlur={formik.handleBlur(ClienteEditFormFields.nomeSocial)}
              w-perm="CLIENTE_INFO_EDIT_NOME_SOCIAL"
            />
            <DatePicker
              label="Data de Nascimento"
              disabled={cliente?.tipo === ClienteTipoEnum.JURIDICA}
              value={formik.values[
                ClienteEditFormFields.dataNascimento
              ].toJSDate()}
              onChange={(date) =>
                formik.setFieldValue(
                  ClienteEditFormFields.dataNascimento,
                  date && new DateOnly(date),
                )
              }
              renderInput={(params) => (
                <TextField
                  {...params}
                  size="small"
                  label="Data de Nascimento"
                  variant="outlined"
                />
              )}
              w-perm="CLIENTE_INFO_EDIT_DATA_NASCIMENTO_IDADE"
            />
            <TextField
              label="Sexo"
              select
              id="sexo"
              size="small"
              variant="outlined"
              value={formik.values.sexo}
              onChange={formik.handleChange(ClienteEditFormFields.sexo)}
              w-perm="CLIENTE_INFO_EDIT_SEXO"
            >
              {OptionsSexo.map((option) => (
                <MenuItem key={option.value} value={option.value}>
                  {option.label}
                </MenuItem>
              ))}
            </TextField>
            <AutocompleteOcupacao
              label="Profissão"
              onChange={(ocupacao) => {
                if (ocupacao)
                  formik.setFieldValue(
                    ClienteEditFormFields.cboId,
                    ocupacao.id,
                  );
              }}
              initialValue={cliente?.cbo}
              w-perm="CLIENTE_INFO_EDIT_PROFISSAO"
            />
            <Box w-perm="CLIENTE_INFO_EDIT_FALECIDO">
              <Typography variant="subtitle2">O cliente é falecido?</Typography>
              <RadioGroupStyled
                name={ClienteEditFormFields.falecido}
                value={String(formik.values.falecido)}
                onChange={(event) => {
                  const value = event.target.value === 'true';
                  formik.setFieldValue(ClienteEditFormFields.falecido, value);
                }}
              >
                <FormControlLabel
                  value="true"
                  control={<Radio color="primary" />}
                  label="Sim"
                />
                <FormControlLabel
                  value="false"
                  control={<Radio color="primary" />}
                  label="Não"
                />
              </RadioGroupStyled>
            </Box>
            <Box w-perm="CLIENTE_INFO_EDIT_DOCUMENTO_INTERNACIONAL">
              <RadioGroupStyled
                name={ClienteEditFormFields.tipoCliente}
                value={String(formik.values.tipoCliente)}
                onChange={formik.handleChange(
                  ClienteEditFormFields.tipoCliente,
                )}
              >
                <FormControlLabel
                  disabled={!!cliente?.documento}
                  value={ClienteTipoEnum.FISICA}
                  control={<Radio color="primary" />}
                  label="CPF"
                />
                <FormControlLabel
                  disabled={!!cliente?.documento}
                  value={ClienteTipoEnum.JURIDICA}
                  control={<Radio color="primary" />}
                  label="CNPJ"
                />
                <FormControlLabel
                  disabled={!!cliente?.documento}
                  value={ClienteTipoEnum.ESTRANGEIRA}
                  control={<Radio color="primary" />}
                  label="ID/Passaporte"
                />
              </RadioGroupStyled>
            </Box>
            <TextFieldStyled
              disabled={!!cliente?.documento}
              id="documento"
              label="CPF/CNPJ/ID ou Passaporte"
              variant="outlined"
              size="small"
              value={formik.values.documento}
              onChange={(value) => handleChangeDocumento(value.target.value)}
              onBlur={formik.handleBlur(ClienteEditFormFields.documento)}
              w-perm="CLIENTE_INFO_EDIT_DOCUMENTO_INTERNACIONAL"
              type={cliente?.documento ? 'text' : typeInput}
              inputProps={{ maxLength }}
            />
          </RenderComponents>
        </StyledForm>
      </CardContentStyled>

      <DialogActionsStyled>
        <ButtonLoadingStyled
          disabled={!formik.dirty}
          onClick={formik.submitForm}
          color="primary"
          variant="contained"
          key="ButtonEditInfoClient"
          size="large"
          isLoading={false}
        >
          Salvar alterações
        </ButtonLoadingStyled>
      </DialogActionsStyled>
    </Card>
  );
}
