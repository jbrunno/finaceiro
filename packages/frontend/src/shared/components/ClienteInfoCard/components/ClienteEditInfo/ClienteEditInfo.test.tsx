import { ClienteSexoEnum, ClienteTipoEnum } from '@/shared/constants';
import { render } from '@test-utils/component-testing';
import { ClienteEditInfo } from './ClienteEditInfo';

const dataMock = new Date('2021-01-01T00:00:00.000Z').toISOString();

const mocks = {
  id: 'teste',
  nome: 'teste',
  nomeSocial: 'teste',
  documento: '11122233344',
  sexo: ClienteSexoEnum.FEMININO,
  dataNascimento: dataMock,
  cbo: {
    id: 'teste',
    codigo: 'teste',
    ocupacao: 'teste',
  },
  falecido: false,
  negativado: false,
  tipo: ClienteTipoEnum.ESTRANGEIRA,
};

jest.mock(
  '@/shared/components/AutocompleteOcupacao/hooks/useAutocompleteOcupacao',
  () => ({
    useAutocompleteOcupacao: () => ({
      handleSelectValue: jest.fn(),
      loadOcupacoes: jest.fn(),
      filterOptions: jest.fn(),
      loading: false,
      label: 'Ocupação',
      ocupacoes: [],
      value: null,
    }),
  }),
);

test('should render ClienteEditInfo', () => {
  const { container } = render(
    <ClienteEditInfo screen="test" cliente={mocks} onSuccess={() => {}} />,
  );
  expect(container).toMatchSnapshot();
});
