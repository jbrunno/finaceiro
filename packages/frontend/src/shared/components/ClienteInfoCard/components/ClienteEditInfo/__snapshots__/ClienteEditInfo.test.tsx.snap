// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClienteEditInfo 1`] = `
.emotion-1 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  overflow: visible;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px -1px rgba(233, 233, 233, 0.8),0px 4px 5px rgba(233, 233, 233, 0.14),0px 1px 10px rgba(233, 233, 233, 0.12);
  background-color: #fff;
}

.emotion-2 {
  padding: 1.5rem 0rem 2rem 0rem;
  text-align: center;
}

.emotion-4 {
  margin: 0;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.8125rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
}

.emotion-6 {
  padding: 16px;
  padding: 0rem 2.5rem 0rem 2.5rem;
}

.emotion-6:last-child {
  padding-bottom: 24px;
}

.emotion-7 {
  display: grid;
  row-gap: 1.5625rem;
  width: 100%;
  min-width: calc(13.125rem + 90px);
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 1rem 2.5rem 3rem 2.5rem;
  width: 100%;
}

.emotion-10>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 8px 22px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #4A31FF;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  text-transform: none;
  width: 100%;
}

.emotion-13::-moz-focus-inner {
  border-style: none;
}

.emotion-13.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-13 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-13:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #2025EF;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-13:hover {
    background-color: #4A31FF;
  }
}

.emotion-13:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-13.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-13.Mui-disabled {
  color: #00000042;
  box-shadow: none;
  background-color: #0000001F;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 emotion-4"
      >
        Informações básicas
      </h3>
    </div>
    <div
      class="MuiCardContent-root emotion-5 emotion-6"
    >
      <form
        class="emotion-7 emotion-8"
      />
    </div>
    <div
      class="MuiDialogActions-root MuiDialogActions-spacing emotion-9 emotion-10"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge emotion-11 emotion-12 emotion-13"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Salvar alterações
      </button>
    </div>
  </div>
</div>
`;
